base/inc/AdePT/BVHNavigator.h

```diff
@@ -158,6 +158,26 @@ class BVHNavigator {
   }
 
 public:
+  // Computes the isotropic safety from the globalpoint.
+  __host__ __device__ static double ComputeSafety(vecgeom::Vector3D<vecgeom::Precision> const &globalpoint,
+                                                  vecgeom::NavStateIndex const &state)
+  {
+    VPlacedVolumePtr_t pvol = state.Top();
+    vecgeom::Transformation3D m;
+    state.TopMatrix(m);
+    vecgeom::Vector3D<vecgeom::Precision> localpoint = m.Transform(globalpoint);
+
+    // need to calc DistanceToOut first
+    vecgeom::Precision safety = pvol->SafetyToOut(localpoint);
+
+    if (safety > 0 && pvol->GetDaughters().size() > 0) {
```

Comment:
Of course the check for the mother safety is needed because it may be the closest boundary, even if not on it. In general, this method does exactly do what @amadio implemented in `BVHSafetyEstimator`.

Reply:
Up to you @hahnjo to apply this change here or not and resolve the conversation, I will leave a general comment in this PR to remember to test if the `safety > tolerance` case gives a real reduction of the full daughter safety checks.

Reply:
I think I would prefer to keep the initial implementation free of micro-optimizations like this. Let's do the fully correct thing for now and check back later if there would be a gain.